session.go

```diff
@@ -848,6 +848,15 @@ func (db *Database) Run(cmd interface{}, result interface{}) error {
 	return db.run(socket, cmd, result)
 }
 
+// RunOnSocket does the same as Run, but guarantees that your command will be run
+// on the provided socket instance; if it's unhealthy, you will receive the error
+// from it.
+func (db *Database) RunOnSocket(socket *mongoSocket, cmd interface{}, result interface{}) error {
```

Comment:
I don't think that accepting a package level parameter in a public method is a good idea. 
There is no way this can be used outside of the mgo package.

Reply:
@KJTsanaktsidis maybe make a private function instead of a public method?

Reply:
Yes, I can definitely do this - don’t think I’ll get to it for a couple of days though, sorry :/

Reply:
@KJTsanaktsidis you could merge https://github.com/zendesk/mgo/pull/1 if you agree with the change

Reply:
Thanks a lot @dvic - I merged your change into this PR. Sorry this took so long for me to get to :/